src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -143,73 +91,79 @@ public void cactoosWay() {
      * Execute 1 task within executor service and ensure that we'll get the
      *  expected exception type.
      */
-    @Test(expected = CompletionException.class)
-    public void executionIsFailedDueToException() {
-        new Threads<String>(
-            Executors.newSingleThreadExecutor(),
-            () -> {
-                throw new IllegalStateException("Something went wrong");
-            }
-        ).iterator().next();
+    @Test
+    public void failsDueToException() {
+        new Assertion<>(
+            "wraps error into CompletionException",
+            () -> new Threads<String>(
+                Executors.newSingleThreadExecutor(),
+                () -> {
+                    throw new IllegalStateException("Something went wrong");
+                }
+            ).iterator().next(),
+            new Throws<>(
+                "java.lang.IllegalStateException: Something went wrong",
+                CompletionException.class
+            )
+        ).affirm();
     }
 
     /**
      * Execute the tasks concurrently using {@link Threads} when
      *  {@link ExecutorService} was initiated by {@link Threads} itself.
      */
     @Test
-    public void cactoosWayWithInlineExecutorService() {
-        this.repeatWithTimeout(
+    public void containsValuesWithInlineExecutorService() {
+        this.repeat(
             arg -> {
-                MatcherAssert.assertThat(
-                    new Threads<String>(
-                        5,
+                new Assertion<>(
+                    // @checkstyle LineLength (1 line)
+                    "contains results from the callables when using the inline executor service",
+                    () -> new Threads<String>(
```

Comment:
Just `Threads<>`

Reply:
@ilyakharlamov build fails due to compiler warning `org/cactoos/experimental/ThreadsTest.java:[121,23] unchecked generic array creation for varargs parameter of type org.cactoos.Scalar<java.lang.String>[]`